src/Currency.php

```diff
@@ -180,17 +179,21 @@ public function format(CurrencyFormatInterface $currencyFormat = null): string
      */
     private function decorate(string $number, CurrencyFormatInterface $currencyFormat): string
     {
+        $separator = '';
+        if (CurrencyFormat::DECORATION_WITH_SPACE === $currencyFormat->getDecorationSpace()) {
+            $separator = ' ';
+        }
```

Comment:
To avoid two assignments, i suggest to change the code to something like this:
```php
$separator = (CurrencyFormat::DECORATION_WITH_SPACE === $currencyFormat->getDecorationSpace())
    ? ' '
    : '';
```